src/coreComponents/constitutive/porosity/PorosityBase.cpp

```diff
@@ -0,0 +1,73 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 Total, S.A
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file PorosityBase.cpp
+ */
+
+#include "PorosityBase.hpp"
+
+namespace geosx
+{
+
+using namespace dataRepository;
+
+namespace constitutive
+{
+
+
+PorosityBase::PorosityBase( string const & name, Group * const parent ):
```

Comment:
I don't think the update is necessary for porosity since I don't see much common interfaces. 

Yes, I was planning to have a poroelastic model that computes porosity as a function of the strain and of the pressure.  This will have its own update class and will be used directly in the SinglePhasePoroelasticKernel to substitute https://github.com/GEOSX/GEOSX/blob/2de5fd5961cbe3d1e4c155ecb04b3c5e47dea06d/src/coreComponents/physicsSolvers/multiphysics/SinglePhasePoroelasticKernel.hpp#L242-254 .